src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -37,6 +42,81 @@ const ChannelListScreenV2 = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestPermission` function is requesting permission and then immediately trying to get the FCM token. However, there's no error handling in case the request for permission fails or the user denies it. It would be better to add some error handling here.
```diff
  const requestPermission = async () => {
    try {
      const authStatus = await messaging().requestPermission();
      const enabled =
        authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL;

      if (enabled) {
        const fcmToken = await messaging().getToken();
        const payload = {
          fcm_token: fcmToken
        };
        fcmTokenService(payload);
      }
    } catch (error) {
      console.error('Failed to request permission or retrieve FCM token:', error);
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->